src/query.rs

```diff
@@ -239,12 +243,40 @@ where
 
         QueryIter {
             types: types.iter(),
-            archetypes: self.archetypes,
+            archetypes: &self.world.archetypes,
             idx: 0,
             len: 0,
             ptr: S::Fetch::null(),
         }
     }
+
+    /// Create a map of the entities matching the query.
+    ///
+    /// This is a faster alternative to [World::get()]/[World::get_mut()].
+    /// For details, see [QueryMap].
```

Comment:
I think it would be better to have the example here and limit the docs of `QueryMap` to the oneliners.

Reply:
I had that first, and then moved it because I was not really sure. So I'll move it back.

Reply:
Done